### PR TITLE
Examples of types of cohesion and coupling

### PR DESCRIPTION
**Pull request by arezookmn**

### CODE REVIEW

File: sources/Arezoo Kamane/Cohesion/LogicalCohesion/UserAuthenticationModule.cs [hunk 1]
```diff
@@ -0,0 +1,68 @@
...
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Cohesion.LogicalCohesion
+{
+    public class UserAuthenticationModule
+    {
+
+        private readonly List<User> registeredUsers;
```
Comment (mohammadKarimi) [inline sources/Arezoo Kamane/Cohesion/LogicalCohesion/UserAuthenticationModule.cs:12]: naming Convention.
private field must be started with _ such as _registeredUsers.

File: sources/Arezoo Kamane/Cohesion/LogicalCohesion/UserAuthenticationModule.cs [hunk 2]
```diff
@@ -0,0 +1,68 @@
...
+
+        public UserAuthenticationModule()
+        {
+            registeredUsers = new List<User>
```
Comment (mohammadKarimi) [inline sources/Arezoo Kamane/Cohesion/LogicalCohesion/UserAuthenticationModule.cs:16]: now you can use collection expression in c#12

File: sources/Arezoo Kamane/Cohesion/LogicalCohesion/UserAuthenticationModule.cs [hunk 3]
```diff
@@ -0,0 +1,68 @@
...
+                new User { Username = "arezoo", Password = "pass123" },
+            };
+        }
+
+        public User AuthenticateUser(string username, string password)
+        {
+            Console.WriteLine("Task: Authenticate User");
+            User user = registeredUsers.FirstOrDefault(u => u.Username == username && u.Password == password);
+
+            if (user != null)
```
Comment (mohammadKarimi) [inline sources/Arezoo Kamane/Cohesion/LogicalCohesion/UserAuthenticationModule.cs:27]: please use pattern matching, is not null

File: sources/Arezoo Kamane/Cohesion/SequentialCohesion/UserState.cs [hunk 4]
```diff
@@ -0,0 +1,15 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Cohesion.SequentialCohesion
```
Comment (mohammadKarimi) [inline sources/Arezoo Kamane/Cohesion/SequentialCohesion/UserState.cs:7]: please use File scoped namespace. 

namespace Cohesion.SequentialCohesion;

File: sources/Arezoo Kamane/Cohesion/SequentialCohesion/UserStateMachine.cs [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Cohesion.SequentialCohesion
+{
+    public class UserStateMachine
+    {
+        private UserState currentState;
```
Comment (mohammadKarimi) [inline sources/Arezoo Kamane/Cohesion/SequentialCohesion/UserStateMachine.cs:11]: Naming Conventions.

File: sources/Arezoo Kamane/Cohesion/SequentialCohesion/UserStateMachine.cs [hunk 6]
```diff
@@ -0,0 +1,49 @@
...
+                Console.WriteLine("Account activation successful. User is now active.");
+                currentState = UserState.Active;
+            }
+            else
+            {
+                Console.WriteLine("Invalid state transition. Account activation can only be performed from 'EmailVerified' state.");
+            }
+        }
+
+        public UserState GetCurrentState()
```
Comment (mohammadKarimi) [inline sources/Arezoo Kamane/Cohesion/SequentialCohesion/UserStateMachine.cs:44]: For methods with single line , it is better to use lambda format. 
public UserState GetCurrentState()
    => currentState;

File: sources/Arezoo Kamane/Cohesion/TemporalCohesion/DailyWorkflow.cs [hunk 7]
```diff
@@ -0,0 +1,32 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
```
Comment (mohammadKarimi) [inline sources/Arezoo Kamane/Cohesion/TemporalCohesion/DailyWorkflow.cs:5]: Please Clean UR usings with ctrl+k+e in Visual studio.

File: sources/Arezoo Kamane/Coupling/CommonCoupling/RegistrationModule.cs [hunk 8]
```diff
@@ -0,0 +1,19 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Coupling.CommonCoupling
+{
+    public class RegistrationModule
+    {
+        public static void RegisterUser(string userName, string email)
```
Comment (mohammadKarimi) [inline sources/Arezoo Kamane/Coupling/CommonCoupling/RegistrationModule.cs:11]: in this code, I can not detect any Common Coupling.

Comment (mohammadKarimi) [inline sources/Arezoo Kamane/Coupling/CommonCoupling/RegistrationModule.cs:11]: username is one word. correct naming is username not userName.

File: sources/Arezoo Kamane/Coupling/CommonCoupling/RegistrationModule.cs [hunk 9]
```diff
@@ -0,0 +1,19 @@
...
+        {
+            int nextUserId = UserRepository.GetAllUsers().Count + 1;
```
Comment (mohammadKarimi) [inline sources/Arezoo Kamane/Coupling/CommonCoupling/RegistrationModule.cs:13]: in this line you might get NullPointerExceptions.

File: sources/Arezoo Kamane/Coupling/CommonCoupling/UserRepository.cs [hunk 10]
```diff
@@ -0,0 +1,28 @@
...
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace Coupling.CommonCoupling
+{
+    public class UserRepository
+    {
+        private static List<User> users = new List<User>();
```
Comment (mohammadKarimi) [inline sources/Arezoo Kamane/Coupling/CommonCoupling/UserRepository.cs:11]: yes, this example is a type of common coupling.

**Review by mohammadKarimi — CHANGES_REQUESTED** [review]
Naming Conventions: please follow these rules. 
Your code is pretty, and complete. but I want you to fix some little issues then send UR PR again.

**Comment by arezookmn** [comment]
Thanks for the comments 
I fixed those issues

**Review by mohammadKarimi — APPROVED** [review]
Perfect.

**Pull request merged**